packages/edit-navigation/src/store/actions.js

```diff
@@ -49,169 +38,311 @@ export function setSelectedMenuId( menuId ) {
  * @param {Object} post A navigation post to process
  * @return {Function} An action creator
  */
-export const createMissingMenuItems = serializeProcessing( function* ( post ) {
+export const createMissingMenuItems = ( post ) => async ( {
+	dispatch,
+	registry,
+} ) => {
 	const menuId = post.meta.menuId;
+	// @TODO: extract locks to a separate package?
+	const lock = await registry
+		.dispatch( 'core' )
```

Comment:
Let's `import { store } from '@wordpress/core-data'` and use it here instead of the `'core'` string. The `./selectors.js` module already does that.